src/Development/IDE/Core/FileStore.hs

```diff
@@ -106,7 +106,10 @@ getModificationTimeRule vfs =
               `catch` \(e :: IOException) -> do
                 let err | isDoesNotExistError e = "File does not exist: " ++ file'
                         | otherwise = "IO error while reading " ++ file' ++ ", " ++ displayException e
-                return (Nothing, ([ideErrorText file $ T.pack err], Nothing))
+                    (fp, _, d) = ideErrorText file (T.pack err)
+                    showDiag = if missingFileDiags then ShowDiag else HideDiag
```

Comment:
Is `HideDiag` really the right thing to use here? So far that is used for diagnostics that we need for things like code actions but don’t want to display as warnings. It seems like this diagnostics is just unused so maybe we should just not produce it?

Reply:
Do we want to filter out _all_ errors or only `isDoesNotExist`?

Reply:
Good points, I'll address both

Reply:
addressed